contrib/vdsm-dev

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+# Author: Albert Esteve <aesteve@redhat.com>
+
+venv_path=$HOME/.venv/vdsm
+venv_activate_path=${venv_path}/bin/activate
+user_storage_path=/var/tmp/vdsm-storage/
+
+print_help()
+{
+    echo "oVirt/vdsm command line helper."
+    echo "Should be run from the repository root folder."
+    echo "Usage: $0 CMD [options]"
+    echo
+    echo "Available commands:"
+    echo "help:          Print this message and exits."
+    echo "build:         Build vdsm and make RPMs."
+    echo "build-install: Build vdsm, make RPMs, and update local packages."
+    echo "test:          Run any test from specified environment using tox."
+    echo "test-storage:  Run one/all storage tests using tox."
+    echo "show-coverage: Show coverage results generated after a test run."
+    echo "               Receives an environment with '-e env'. [default=storage]"
+}
+
+venv_create()
+{
+    echo "Creating venv at $venv_path"
+    python3 -m venv ${venv_path}
+    source "${venv_activate_path}"
+    pip install --upgrade pip
+    pip install -r docker/requirements.txt
+    deactivate
+}
+
+make_storage()
+{
+    if [ ! -d $user_storage_path ]; then
+        echo "Make storage at $user_storage_path"
+        make storage
+    fi
+}
+
+test_path()
+{
+    path=$1
+    storage_prefix=storage/
+    if [ ! -f "tests/$path" ]; then
+        path=${storage_prefix}${path}
+        if [ ! -f "tests/$path" ]; then
+            echo "No test $path found"
+            exit 1
+        fi
+    fi
+}
+
+vdsm_build()
+{
+    git clean -dxf && ./autogen.sh --system --enable-timestamp && \
+         make && rm -rf ~/rpmbuild && make rpm
+}
+
+handle_test_opts()
+{
+    while test $# -gt 0
+    do
+        case "$1" in
+            -e)
+                shift
+                env=$1
+                if [ $env == "storage" ]; then
+                    make_storage
+                fi
+                ;;
+            *)
+                # Let tox handle the rest of the arguments
+                ;;
+        esac
+        shift
+    done
+}
```

Comment:
Since it does shift inside and all, I think it is actually easier to handle if it is kept separately.

But I have added some comment to clarify the purpose and renamed the function.